### PR TITLE
PKG: Declare Pillow as optional dependency

### LINKED ISSUE #1390: PyPDF2 doesn't install pillow
Running PyPDF2 from scratch and trying to extract images (via `page.images` property) throws ModuleNotFoundError.

image

This is because my current setup doesn't have pillow installed, _and PyPDF2 doesn't install it_. My proposal is to either add an extra to the setuptools config that installs pillow, or to have this requirement more visible in the docs. 

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-12.6-x86_64-i386-64bit

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.11.1
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
with open("2210.03747.pdf", "rb") as your_pdf_file:
    pdf = PdfReader(your_pdf_file)
    for page in pdf.pages:
        for image in page.images:
            print(image)
```

Any PDF with xObject images; for example: https://arxiv.org/pdf/2210.03747.pdf

## Traceback

This is the complete Traceback I see:

```
x_object_obj = {'/Type': '/XObject', '/Subtype': '/Image', '/Width': 417, '/Height': 328, '/BitsPerComponent': 8, '/ColorSpace': '/DeviceRGB', '/SMask': IndirectObject(33, 0, 4575023296), '/Filter': '/FlateDecode'}

    def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
        """
        Users need to have the pillow package installed.
    
        It's unclear if PyPDF2 will keep this function here, hence it's private.
        It might get removed at any point.
    
        :return: Tuple[file extension, bytes]
        """
>       from PIL import Image
E       ModuleNotFoundError: No module named 'PIL'

venv/lib/python3.10/site-packages/PyPDF2/filters.py:567: ModuleNotFoundError
```

### PR DESCRIPTION
**Pull request by MartinThoma**
Closes #1390

### CODE REVIEW

**Comment by MartinThoma** [comment]
@MasterOdin What do you think about this? I think (hope) that we don't get more dependencies, but with `full` people would have a clear way how to install all dependencies.

I don't see a way around the PIL / cryptodome dependencies if we want to have those features. And I think those features add enough value to justify an optional dependency.

I am uncertain about the name "full". Maybe "all" would be better?

**Review by MasterOdin — CHANGES_REQUESTED** [review]
Should include a link on https://github.com/py-pdf/PyPDF2/blob/main/docs/user/extract-images.md back to the installation page referencing that need to install `[image]`.

May also want to update https://pypdf2.readthedocs.io/en/latest/modules/PageObject.html#PyPDF2._page.PageObject.images to include that reference as well?

File: docs/user/encryption-decryption.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Encryption and Decryption of PDFs
 
 Please see the note in the
-[Installation doc](https://pypdf2.readthedocs.io/en/latest/user/installation.html)
+[Installation doc](installation.html)
 for installing the extra dependencies if interacting with PDFs that use AES.
```
Comment (MasterOdin) [inline docs/user/encryption-decryption.md:5]: Should this be modified to have the `>` prefix?

**Review by MasterOdin — APPROVED** [review]

File: PyPDF2/filters.py [hunk 2]
```diff
@@ -564,7 +564,13 @@ def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
...
     :return: Tuple[file extension, bytes]
     """
-    from PIL import Image
+    try:
+        from PIL import Image
+    except ImportError:
+        raise ImportError(
+            "pillow is required to do image extraction. "
+            "It can be installed via 'pip install PyPDF2[image]'"
+        )
```
Comment (MasterOdin) [inline PyPDF2/filters.py:573]: If wanted code coverage here, could add a test with `monkeypatch.setitem` modifying `sys.modules` to remove it from the environment. See https://stackoverflow.com/a/28361013/4616655 for an example.

Comment (MartinThoma) [reply PyPDF2/filters.py:573]: That's an awesome idea! Thanks for sharing it - I've written the test!

**Pull request merged**